src/programs/sum_all_eer_files/sum_all_eer_files.cpp

```diff
@@ -0,0 +1,215 @@
+    output_sum_image.SetToConstant(0.0);
+
+    if ( make_dark_and_gain == true ) {
+        output_dark_image.Allocate(file_x_size, file_y_size, 1);
+        output_gain_image.Allocate(file_x_size, file_y_size, 1);
+    }
+
+    global_sum_image = new double[output_sum_image.real_memory_allocated];
+    ZeroDoubleArray(global_sum_image, output_sum_image.real_memory_allocated);
+
+    if ( make_dark_and_gain == true ) {
+        global_sum_squares_image = new double[output_sum_image.real_memory_allocated];
+        ZeroDoubleArray(global_sum_squares_image, output_sum_image.real_memory_allocated);
+    }
+
+    // loop over all files, and do summing..
+
+    wxPrintf("Summing All Files...\n\n");
+    ProgressBar* my_progress = new ProgressBar(all_files.GetCount( ));
+
+    int number_processed = 0;
+    // thread if available
+#pragma omp parallel default(none) num_threads(max_threads) shared(make_dark_and_gain, global_sum_image, global_sum_squares_image, output_sum_image, all_files, total_summed, number_processed, my_progress, eer_super_res_factor, eer_frames_per_image) private(file_counter, current_input_file, frame_counter, sum_image, sum_squares_image, pixel_counter)
+    { // bracket for omp
+
+        Image buffer_image;
+        int   my_total_summed = 0;
+        sum_image             = new double[output_sum_image.real_memory_allocated];
+        ZeroDoubleArray(sum_image, output_sum_image.real_memory_allocated);
+
+        if ( make_dark_and_gain == true ) {
+            sum_squares_image = new double[output_sum_image.real_memory_allocated];
+            ZeroDoubleArray(sum_squares_image, output_sum_image.real_memory_allocated);
+        }
+
+#pragma omp for
+        for ( file_counter = 0; file_counter < all_files.GetCount( ); file_counter++ ) {
+            //wxPrintf("Summing file %s...\n", all_files.Item(file_counter));
+        	wxPrintf("Summing File %ld\n", file_counter);
+        	current_input_file = new EerFile();
+        	current_input_file->OpenFile(all_files.Item(file_counter).ToStdString( ), false, false, false, eer_super_res_factor, eer_frames_per_image);
+
+            for ( frame_counter = 0; frame_counter < current_input_file->ReturnNumberOfSlices( ); frame_counter++ ) {
+                buffer_image.ReadSlice(current_input_file, frame_counter + 1);
+
+                my_total_summed++;
+
+                for ( pixel_counter = 0; pixel_counter < output_sum_image.real_memory_allocated; pixel_counter++ ) {
+                    sum_image[pixel_counter] += buffer_image.real_values[pixel_counter];
+
+                    if ( make_dark_and_gain == true ) {
+                        sum_squares_image[pixel_counter] += pow(buffer_image.real_values[pixel_counter], 2);
+                    }
+                }
+                //sum_image.AddImage(&buffer_image);
+            }
+
+            current_input_file->CloseFile( );
+            delete current_input_file;
+
+#pragma omp atomic
```

Comment:
I realize looking at this that this isn't something you changed from the other sum_alls*

But, the omp atomic clause only covers the update to the counter, while the call to the shared object - my_progress-> update - is not protected as it would be in a scoped omp critical.

I know the atomic is more efficient, but wouldn't this be exposing race conditions on the progress bar update @timothygrant80 ?

Reply:
Given that this is not unique to this PR, I'll submit a separate issue and mark this comment resolved.